packages/vscode-extension/src/webview/components/InspectDataMenu.tsx

```diff
@@ -1,10 +1,48 @@
+import React, { useState } from "react";
 import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
 import { Frame, InspectDataStackItem } from "../../common/Project";
 import { DeviceProperties } from "../utilities/consts";
 import "./InspectDataMenu.css";
 
 type OnSelectedCallback = (item: InspectDataStackItem) => void;
 
+const MAX_INSPECT_ITEMS = 5;
+
+interface InspectItemProps {
+  item: InspectDataStackItem;
+  onSelected: (item: any) => void;
+  onHover: (item: any) => void;
+}
+
+const InspectItem = React.forwardRef<HTMLDivElement, InspectItemProps>(
+  ({ item, onSelected, onHover }, forwardedRef) => {
+    const extractFileDetails = (filePath: string) => {
```

Comment:
Imo, now you don't need a helper (I meant component or helper, but I prefer the component) - just fyi, when you define a method in render, it's created on every rerender